File: backend/Models/ServiceItem.cs
using System.ComponentModel.DataAnnotations;

namespace backend.Models;

public class ServiceItem
{
    [Key]
    public Guid Id { get; set; }
    public Guid WorkspaceId { get; set; }
    public string Name { get; set; } = string.Empty;
    public string? Description { get; set; }
    public ServiceItemType Type { get; set; } = ServiceItemType.Service;

    public string Category { get; set; } = string.Empty;
    public string? SKU { get; set; }
    [Range(0, double.MaxValue)]
    public decimal UnitPrice { get; set; }
    [Range(0, double.MaxValue)]
    public decimal Cost { get; set; }
    public bool IsTaxable { get; set; }
    public bool IsActive { get; set; }
    public string? ImageUrl { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
}

public enum ServiceItemType
{
    Service,
    Material
}

File: frontend/src/types/ServiceItem.ts
import { ServiceItemType } from '../constants/Enumeration/ServiceItem/ServiceItem';

export type TServiceItem = {
  id: string;
  workspaceId: string;
  name: string;
  description?: string;
  type: ServiceItemType;
  category: string;
  sku: string;
  unitOfMeasure: string;
  stockLevel: number;
  reorderPoint: number;
  defaultDurationMinutes?: number;
  markupPercentage: number;
  vendor?: string;
  unitPrice: number;
  cost: number;
  taxRate?: number;
  isTaxable: boolean;
  isActive: boolean;
  imageUrl?: string;
  createdAt: string;
  updatedAt: string;
};

export type TAddServiceItem = {
  workspaceId: string;
  name: string;
  description?: string;
  type: ServiceItemType;
  category: string;
  sku?: string;
  unitOfMeasure: string;
  stockLevel?: number;
  reorderPoint?: number;
  defaultDurationMinutes?: number;
  markupPercentage?: number;
  vendor?: string;
  unitPrice: number;
  cost: number;
  taxRate?: number;
  isTaxable: boolean;
  isActive: boolean;
  imageUrl?: string;
};

export type TUpdateServiceItem = {
  id: string;
  name?: string;
  description?: string;
  type?: ServiceItemType;
  category?: string;
  sku?: string;
  unitOfMeasure?: string;
  stockLevel?: number;
  reorderPoint?: number;
  defaultDurationMinutes?: number;
  markupPercentage?: number;
  vendor?: string;
  unitPrice?: number;
  cost?: number;
  taxRate?: number;
  isTaxable?: boolean;
  isActive?: boolean;
  imageUrl?: string;
};

export type TImportServiceItem = {
  workspaceId: string;
  name?: string;
  description?: string;
  type?: ServiceItemType;
  category?: string;
  sku?: string;
  unitOfMeasure?: string;
  stockLevel?: number;
  reorderPoint?: number;
  defaultDurationMinutes?: number;
  markupPercentage?: number;
  vendor?: string;
  unitPrice?: number;
  cost?: number;
  taxRate?: number;
  isTaxable?: boolean;
  isActive?: boolean;
  imageUrl?: string;
};

export type TServiceItemFilter = {
  q?: string;
  sortBy?: string;
  sort?: string;
  category?: string;
  priceMin?: number;
  priceMax?: number;
  isActive?: boolean;
  hasImage?: boolean;
  description?: string;
};

export type TServiceItemStats = {
  totalItems: number;
  totalMaterialItems: number;
  totalServiceItems: number;
  totalPricebookValue: number;
  averageItemPrice: number;

  totalItemsChange: string;
  materialItemsChange: string;
  serviceItemsChange: string;
  averageItemPriceChange: string;
};

export type TSortOption = {
  id: string;
  label: string;
  sortBy: string;
  sort: string;
};
